src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,6 +171,11 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
+        public static void RequiresCanRead(Expression expression, string paramName)
+        {
+            RequiresCanRead(expression, paramName, -1);
+        }
```

Comment:
Maybe. I think it'd take a lot of work and have little return. Taking a look at the code for RequiresCanRead, we pass the index to the exception throwing helpers: https://github.com/hughbe/corefx/blob/4f2fa6014e82ea87ffee20150cc0ce22d0a194ae/src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs#L190

This pattern looks unavoidable